tests/Nox.Types.Tests/Types/LanguageCode/LanguageCodeTests.cs

```diff
@@ -1,11 +1,66 @@
-﻿// ReSharper disable once CheckNamespace
-namespace Nox.Types.Tests.Types;
+﻿
+using FluentAssertions;
+using Nox.Types.Tests.Types.LanguageCode;
+
+namespace Nox.Types.Tests;
 
 public class LanguageCodeTests
 {
+    [Theory]
+    [ClassData(typeof(LanguageCodeTestsDataClass))]
+    public void CreatingLanguageCode_From_IsValid(string languageCodeString)
+    {
+        var languageCode = LanguageCode.From(languageCodeString);
+
+        languageCode.Value.Should().Be(languageCodeString);
+    }
+
+    [Fact]
+    public void CreatingLanguageCode_UnsupportedLanguageCode_ThrowsValidationException()
+    {
+        var action = () => LanguageCode.From("ABC");
+
+        action.Should().Throw<TypeValidationException>()
+            .And.Errors.Should().BeEquivalentTo(new[] { new ValidationFailure("Value", "Could not create a Nox LanguageCode type with unsupported value 'ABC'.") });
+    }
+
+    [Fact]
+    public void CreatingLanguageCode_SameLanguageCode_ReturnsEqual()
+    {
+        var languageCode1 = LanguageCode.From("EN");
+
+        var languageCode2 = LanguageCode.From("EN");
+
+        languageCode2.Should().Be(languageCode1);
+    }
+
+
+    [Fact]
+    public void CreatingLanguageCode_CaseDifferentLanguageCode_ReturnsEqual()
+    {
```

Comment:
The class correctly allows case-insensitive creation of LanguageCode, but doesn't do a case insensitive comparison. I suggest we always internally convert language code Value to Invariant lower case, and change the Regex to be case sensitive.